File: 18258.cpp
#pragma warning(disable:4996)
#include <iostream>
#include <string>
#include <queue>
using namespace std;

int main() {

	cin.tie(NULL);
	cin.sync_with_stdio(false);

	int N;
	cin >> N;
	string str;
	queue<int> q;
	int num;


	for (int i = 0; i < N; i++) {
		cin >> str;
		if (str == "push") {
			cin >> num;
			q.push(num);
		}
		if (str == "pop") {
			if (q.empty()) {
				cout << "-1\n";
				continue;
			}
			cout << q.front() << "\n";
			q.pop();
		}
		if (str == "size")
			cout << q.size() << "\n";
		if (str == "empty")
			cout << q.empty() << "\n";
		if (str == "front") {
			if (q.empty())
				cout << "-1\n";
			else
				cout << q.front() << "\n";
		}
		if (str == "back") {
			if (q.empty())
				cout << "-1\n";
			else
				cout << q.back() << "\n";
		}
	}



	return 0;
}

File: 18258.py
import sys
from collections import deque
input = sys.stdin.readline

N = int(input())

queue = deque()
top = -1
for _ in range(N):
    order = input()
    
    if "push" in order:
        ord, num = order.split()
        queue.append(num)
        top += 1
    elif "pop" in order:
        if top == -1:
            print("-1")
        else:
            print(queue.popleft())
            top -= 1
    elif "size" in order:
        print(top+1)
    elif "empty" in order:
        if top == -1:
            print("1")
        else:
            print("0")
    elif "front" in order:
        if top == -1:
            print("-1")
        else:
            print(queue[0])
    elif "back" in order:
        if top == -1:
            print("-1")
        else:
            print(queue[top])
